common/reviews/api/heft.api.md

```diff
@@ -248,8 +249,9 @@ export interface IHeftActionConfigurationOptions {
 }
 
 // @beta
-export interface IHeftBaseParameter {
+export interface IHeftBaseParameter<TValue> {
```

Comment:
`<TValue>` enables us to delete a few lines of code.  But it seems to hurt the experience for a consumer of the API:

- Now you have to follow a generic parameter to figure out what the value is
- We can't write specific `/** */` docs for each kind of `value` (when that field is the centerpiece of this data structure)
- If more parameter-specific data is exposed in the future, those `type IHeftIntegerParameter =` declarations will have to be converted back to interfaces anyway.

I think I prefer @halfnibble's more verbose design.

Reply:
> The only API change is changing the value property of the string list parameter's name from `values` to `value` to be consistent with the other parameter types. This deviates from `ts-command-line`, so it probably warrants some discussion.

As mentioned previously, the overhaul of `ts-command-line` will give us a chance to rethink some of these API decisions.  But the changes need to start with:  *"What problem is this solving?"*

Reply:
We are leaving this as .value for now and will have a discussion further down the road.